adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/ConnectionLimitManager.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.amqp;
+
+import org.eclipse.hono.config.ProtocolAdapterProperties;
+import org.eclipse.hono.service.metric.Metrics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class manages a limit of concurrent connections. The limit can be configured in
+ * {@link ProtocolAdapterProperties#setMaxConnections(int)}. If no value is configured, it will set a limit based on the
+ * given maximum amount of memory and the AMQP session window size (
+ * {@link org.eclipse.hono.adapter.amqp.AmqpAdapterProperties#getMaxSessionWindowSize()}).
+ * 
+ */
+public class ConnectionLimitManager {
```

Comment:
Hmmm, we already use something similar for the MQTT adapter, don't we? This smells like a strategy interface with different implementations for different protocol adapters, don't you think?

Reply:
Absolutely. I thought of doing the alignment between both modules (and moving common code to module `service-base`) in a separate PR. But I can do it here if you prefer.

Reply:
Yes, I would rather have it in this PR and not add code that we immediately remove again.